src/shared/sam/cli/samCliInvoker.ts

```diff
@@ -8,42 +8,168 @@
 import { SpawnOptions } from 'child_process'
 import * as vscode from 'vscode'
 import { extensionSettingsPrefix } from '../../constants'
+import { getLogger, Logger } from '../../logger'
 import { DefaultSettingsConfiguration } from '../../settingsConfiguration'
 import { ChildProcess, ChildProcessResult } from '../../utilities/childProcess'
 import { DefaultSamCliConfiguration, SamCliConfiguration } from './samCliConfiguration'
+import {
+    DefaultSamCliUtils,
+    SamCliInfoResponse,
+    SamCliProcessInfo,
+    SamCliProcessInvoker,
+    SamCliTaskInvoker,
+    SamCliUtils } from './samCliInvokerUtils'
 import { DefaultSamCliLocationProvider } from './samCliLocator'
+import {
+    DefaultSamCliVersionValidator,
+    SamCliVersionValidation,
+    SamCliVersionValidator,
+    SamCliVersionValidatorResult
+} from './samCliVersionValidator'
 
-export interface SamCliProcessInvoker {
-    invoke(options: SpawnOptions, ...args: string[]): Promise<ChildProcessResult>
-    invoke(...args: string[]): Promise<ChildProcessResult>
+interface SamCliProcessInvokerContext {
+    cliConfig: SamCliConfiguration
+    cliInfo: SamCliProcessInfo
+    cliUtils: SamCliUtils
+    validator: SamCliVersionValidator
 }
 
-export interface SamCliTaskInvoker {
-    invoke(task: vscode.Task): Promise<vscode.TaskExecution>
+export function makeSamCliProcessInvokerContext(
```

Comment:
Also re consistency: Elsewhere in our codebase, we define a class `DefaultFooContext`. Unless there's a good reason to avoid it, can you use that style?